specutils/wcs/adapters/fitswcs_adapter.py

```diff
@@ -71,17 +81,18 @@ def spec_axis(self):
         """
         self._spec_axis = self.wcs.wcs.spec
 
-        if self._spec_axis < 0:
-            try:
-                idx = list(self.wcs.wcs.ctype).index('LINEAR')
-            except ValueError:
+        if (self._spec_axis < 0) and (self._wcs.wcs.spec) < 0:
+            ctypelist = list(self.wcs.wcs.ctype)
+            for n in self.substitute_spec_axis_names:
```

Comment:
I think it might best to use regex, or at least perhaps only checking in lower case in order to avoid issues where the axis is defined as `linear` or `WAVELENGTH` (since currently, the casing has to be absolutely correct).

Reply:
Done. (In a different line, thus github does not know this is resolved.)